file/builder.go

```diff
@@ -598,13 +602,31 @@ func (b *stateBuilder) routes() {
 		return
 	}
 
+	unsupportedRoutes := []string{}
 	for _, r := range b.targetContent.Routes {
 		r := r
 		if err := b.ingestRoute(r); err != nil {
 			b.err = err
 			return
 		}
 	}
+
+	// check routes' paths format
+	if b.checkRoutePaths {
+		allRoutes, err := b.intermediate.Routes.GetAll()
+		if err != nil {
+			b.err = err
+			return
+		}
+		for _, r := range allRoutes {
+			if utils.HasPathsWithRegex300AndAbove(r.Route) {
+				unsupportedRoutes = append(unsupportedRoutes, *r.Route.ID)
+			}
+		}
+	}
+	if len(unsupportedRoutes) > 0 {
+		utils.PrintRouteRegexWarning()
```

Comment:
Why not print first 10 names/id?

Reply:
Once I started messing with many names, I decided that maybe it was better to not screw up message lines length too much. If you prefer to include them, I'll put them back.